extensions/oidc/runtime/src/main/java/io/quarkus/oidc/VertxKeycloakRecorder.java

```diff
@@ -62,4 +62,23 @@ public void handle(AsyncResult<OAuth2Auth> event) {
         mechanism.setAuth(auth);
         mechanism.setAuthServerURI(config.authServerUrl);
     }
+
+    protected static OIDCException toOidcException(Throwable cause) {
+        final String message = "OIDC server is not available at the 'quarkus.oidc.auth-server-url' URL. "
+                + "Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example:"
+                + " 'https://localhost:8180/auth/realms/quarkus'";
+        Throwable vertxCause = findVertxCause(cause);
```

Comment:
I think findVertxCause has the potential to do more harm than good. If vert.x throws for example an IllegalStateException then the user will get no information. Just including the more informative error message should be fine.

Reply:
@stuartwdouglas but is there a scope for an exception like `IllegalStateException` in this particular case :-) ? The only other exception I can think of is the connect exception, let me check what happens in this case.
I don't mind really dropping this finder if it is of a concern...

Reply:
@stuartwdouglas I haven't found an obvious difference in the way the stacktrace is reported with or without `findVertxCause`, it looked like it was more verbose without it, but in both cases I could find the root Vertx message, with the descriptive message starting on top. So I've just dropped `findVertxCause`. I guess if the users will start saying it is difficult to see through the trace then we can revisit, thanks